core/src/main/java/org/apache/druid/math/expr/Function.java

```diff
@@ -3275,4 +3276,86 @@ public ExprEval apply(List<Expr> args, Expr.ObjectBinding bindings)
       return l.stream();
     }
   }
+
+  abstract class SizeFormatFunc implements Function
+  {
+    protected abstract HumanReadableBytes.UnitSystem getUnitSystem();
+
+    /**
+     * Evaluate given expression
+     * By default, 'precision' is 2 and 'hasSpace' is false
+     */
+    @Override
+    public ExprEval apply(List<Expr> args, Expr.ObjectBinding bindings)
+    {
+      final long bytes = args.get(0).eval(bindings).asLong();
```

Comment:
Thinking out loud, how should this function behave with non-long inputs? The way this is currently implement:
* Inputs of `ExprType.DOUBLE` will be cast to a `ExprType.LONG` before conversion.
* For `ExprType.STRING` inputs, if they are number-ish strings, they will be parsed into long values, but if not will always be 0 if they are not parseable as numbers, since we are calling `asLong` without checking `isNumericNull` of the `ExprEval`, so we are ignoring the value of `druid.generic.useDefaultValueForNull`.

I don't know that this behavior is incorrect, I just wanted to call it out to think about it. I do think we want to check for `isNumericNull` and return `ExprEval.of(null)` if `NullHandling.sqlCompatible()` is set.

I see in the SQL operator it looks like it strictly validates that the inputs are numeric, while Druid native expressions have traditionally been a bit fast and loose about the inputs they accept and tend to be rather forgiving, so perhaps this is ok that the behavior here doesn't quite match.

Reply:
good point. I'll make some improvement here.

Reply:
Hi @clintropolis , null and type handling has been improved in the latest commit. Please check it at your convenience.